x/ibc/23-commitment/merkle.go

```diff
@@ -106,10 +103,11 @@ func (p Path) String() string {
 // CONTRACT: provided path string MUST be a well formated path. See ICS24 for
 // reference.
 func ApplyPrefix(prefix PrefixI, path string) (Path, error) {
-	err := host.DefaultPathValidator(path)
-	if err != nil {
-		return Path{}, err
-	}
+	// TODO: fix path check
```

Comment:
what is failing actually failing here?

Reply:
yeah, `host.DefaultPathValidator` will check the path `isAlphaNumeric`, reality it has the prefix `[]byte{0x00}`. So I wonder if we can revert the prefix using submodule_name, such as `connection`, `channel`, etc.
```
func DefaultPathValidator(path string) error {
	pathArr := strings.Split(path, "/")
	for _, p := range pathArr {
		// Each path element must either be alphanumeric
		if !isAlphaNumeric(p) {
			return sdkerrors.Wrapf(ErrInvalidPath(DefaultCodespace, path), "invalid path element containing non-alphanumeric characters: %s", p)
		}
	}
	return nil
}
```